lib/ex_unit/lib/ex_unit/capture_server.ex

```diff
@@ -83,25 +84,71 @@ defmodule ExUnit.CaptureServer do
     {:noreply, config}
   end
 
-  defp release_device(ref, %{devices: {names, refs}} = config) do
-    case Map.pop(refs, ref) do
-      {{name, pid}, refs} ->
-        names = Map.delete(names, name)
-        Process.demonitor(ref, [:flush])
-
-        try do
-          try do
-            Process.unregister(name)
-          after
-            Process.register(pid, name)
-          end
-        rescue
-          ArgumentError -> nil
-        end
+  defp capture_device(name, encoding, "", %{devices: devices} = config, caller)
+       when :erlang.is_map_key(name, devices) do
+    case Map.get(devices, name) do
+      %{encoding: ^encoding} = device ->
+        {_, output} = StringIO.contents(device.pid)
+        ref = Process.monitor(caller)
+        refs = [{ref, byte_size(output)} | device.refs]
+        devices = %{devices | name => %{device | refs: refs}}
```

Comment:
This can be written as:

    update_in config.devices[name].refs, &[{ref, byte_size(output)} | &1]

And if you make the refs a map, as proposed below, it can be written as:

    put_in config.devices[name].refs[ref], byte_size(output)